test/mcmc/gibbs.jl

```diff
@@ -281,56 +445,85 @@ end
                 m[k] ~ Normal(1.0, 1.0)
             end
         end
-        model = imm(Random.randn(100), 1.0)
+        num_zs = 100
+        num_samples = 10_000
+        model = imm(Random.randn(num_zs), 1.0)
         # https://github.com/TuringLang/Turing.jl/issues/1725
         # sample(model, Gibbs(; z=MH(), m=HMC(0.01, 4)), 100);
-        sample(model, Gibbs(; z=PG(10), m=HMC(0.01, 4; adtype=adbackend)), 100)
+        chn = sample(
+            model, Gibbs(; z=PG(10), m=HMC(0.01, 4; adtype=adbackend)), num_samples
+        )
+        # The number of m variables that have a non-zero value in a sample.
+        num_ms = count(ismissing.(Array(chn[:, (num_zs + 1):end, 1])); dims=2)
+        # The below are regression tests. The values we are comparing against are from
+        # running the above model on the "old" Gibbs sampler that was in place still on
+        # 2024-11-20. The model was run 5 times with 10_000 samples each time. The values
+        # to compare to are the mean of those 5 runs, atol is roughly estimated from the
+        # standard deviation of those 5 runs.
+        # TODO(mhauru) Could we do something smarter here? Maybe a dynamic model for which
+        # the posterior is analytically known? Doing 10_000 samples to run the test suite
+        # is not ideal
+        @test isapprox(mean(num_ms), 8.6087; atol=0.5)
```

Comment:
Maybe add an issue for future ref?

Reply:
Opened here: https://github.com/TuringLang/Turing.jl/issues/2402. Will also add comments in the code referencing this issue and the other one I just opened.